### PR TITLE
Add type stubs & minimal validation

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This adds type stubs to the published package, so that consumers can benefit from type checking despite the Cython implementation of this package. The stubs are manually constructed as there does not appear to be a mature solution which produces good stubs from Cython code. (It is actually unclear if such a solution is possible without annotation of the Cython source)

Validation of the stubs is achieved by annotating the tests and relying on the existing test coverage to explore all relevant members & operations. A new tox environment runs mypy against the tests.

Fixes https://github.com/Ezibenroc/PyRoaringBitMap/issues/105

Builds on #109.

### CODE REVIEW

**Comment by Ezibenroc** [comment]
Great work!

> The stubs are manually constructed as there does not appear to be a mature solution which produces good stubs from Cython code.

That's kind of crazy that there is nothing to automate this stub creation.

**Comment by PeterJCLaw** [comment]
> That's kind of crazy that there is nothing to automate this stub creation.

There's work ongoing it seems, but there didn't seem to be a default just-works solution. I didn't look too much into all the options, partly as some of them looked like they'd be fairly disruptive. https://github.com/mike-huls/cythonbuilder for example seems to be mentioned as being able to do it, but also seems to be a separate build process -- which I didn't want to force on the project just for annotations. I didn't look at all into how it actually works though, nor whether it can be used just for the stubs.

**Comment by Ezibenroc** [comment]
Alright!
The PR is still marked as "work-in-progress", do you still have changes to do before I review?

I am working on #106, adding support to 64-bit bitmaps. I feel it might be easier to first merge #110 and then #106, rather than the other way around. What do you think?

**Comment by PeterJCLaw** [comment]
It's mostly that I'll rebase this on `master` once #109 is merged, so that the diff is clearer. I also wanted to self-review the types added to the tests.

I haven't looked at #106 in detail, though it seems that that adds quite a lot of stuff? If so, yeah it's probably simpler to get this PR in first. Hopefully extending the changes here to cover that will be fairly straightforward.

**Comment by Ezibenroc** [comment]
I just merged #109, tell me when you are ready for review :+1:

File: pyroaring/__init__.pyi [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    n_values_array_containers: int
+    n_values_run_containers: int
+    n_values_bitset_containers: int
+    n_bytes_array_containers: int
+    n_bytes_run_containers: int
+    n_bytes_bitset_containers: int
+    max_value: int
+    min_value: int
+    sum_value: int
+    cardinality: int
```
Comment (Ezibenroc) [inline pyroaring/__init__.pyi:28]: That made me think, it could be nicer to replace this dict by a proper dataclass. But this is out of scope of this PR of course.

**Comment by PeterJCLaw** [comment]
For clarity: this is ready for review now.

**Review by Ezibenroc — APPROVED** [review]
